languages/thingtalk/utils.js

```diff
@@ -53,7 +53,10 @@ function makeFilter(loader, pname, op, value, negate = false) {
     if (ptype.isEntity && ptype.type === 'tt:url')
         return null;
     if (op === 'contains') {
-        ptype = Type.Array(vtype);
+        if (loader.params.out.has(pname.name + '+' + Type.RecurrentTimeSpecification))
+            ptype = Type.RecurrentTimeSpecification;
+        else
+            ptype = Type.Array(vtype);
```

Comment:
This is a little bit bad, once you have multiple domains using different types for the same parameter name.

Reply:
Not sure what's the best soln, but I added another constaint checking if vtype is date or time